frame/system/benchmarking/src/lib.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+// Benchmarks for Utility Pallet
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::Encode;
+use sp_std::vec;
+use sp_std::prelude::*;
+use sp_core::{ChangesTrieConfiguration, storage::well_known_keys};
+use sp_runtime::traits::Hash;
+use frame_benchmarking::{benchmarks, account};
+use frame_support::storage::{self, StorageMap};
+use frame_system::{Module as System, Call, RawOrigin, DigestItemOf, AccountInfo};
+
+mod mock;
+
+const SEED: u32 = 0;
+
+pub struct Module<T: Trait>(System<T>);
+pub trait Trait: frame_system::Trait {}
+
+benchmarks! {
+	_ { }
+
+	remark {
+		// # of Bytes
+		let b in 0 .. 16_384;
+		let remark_message = vec![1; b as usize];
+		let caller = account("caller", 0, SEED);
+	}: _(RawOrigin::Signed(caller), remark_message)
+
+	set_heap_pages {
+		// Heap page size
+		let i in 0 .. u32::max_value();
+	}: _(RawOrigin::Root, i.into())
+
+	// `set_code` was not benchmarked because it is pretty hard to come up with a real
+	// Wasm runtime to test the upgrade with. But this is okay because we will make
+	// `set_code` take a full block anyway.
+
+	set_code_without_checks {
+		// Version number
+		let b in 0 .. 16_384;
+		let code = vec![1; b as usize];
+	}: _(RawOrigin::Root, code)
+	verify {
+		let current_code = storage::unhashed::get_raw(well_known_keys::CODE).ok_or("Code not stored.")?;
+		assert_eq!(current_code.len(), b as usize);
+	}
+
+	set_changes_trie_config {
+		let d in 0 .. 1000;
+
+		let digest_item = DigestItemOf::<T>::Other(vec![]);
+
+		for i in 0 .. d {
+			System::<T>::deposit_log(digest_item.clone());
+		}
+		let changes_trie_config = ChangesTrieConfiguration {
+			digest_interval: d,
+			digest_levels: d,
+		};
+	}: _(RawOrigin::Root, Some(changes_trie_config))
+	verify {
```

Comment:
Do you want to verify that the trie config ends up in storage?

Reply:
if a log is added, all other parts of the code has run successfully:

```
		#[weight = (20_000_000, DispatchClass::Operational)]
		pub fn set_changes_trie_config(origin, changes_trie_config: Option<ChangesTrieConfiguration>) {
			ensure_root(origin)?;
			match changes_trie_config.clone() {
				Some(changes_trie_config) => storage::unhashed::put_raw(
					well_known_keys::CHANGES_TRIE_CONFIG,
					&changes_trie_config.encode(),
				),
				None => storage::unhashed::kill(well_known_keys::CHANGES_TRIE_CONFIG),
			}

			let log = generic::DigestItem::ChangesTrieSignal(
				generic::ChangesTrieSignal::NewConfiguration(changes_trie_config),
			);
			Self::deposit_log(log.into());
		}